ClientUpdater/Updater.cs

```diff
@@ -15,1140 +15,1152 @@
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+namespace ClientUpdater;
+
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
 using System.Net;
-using System.Net.Cache;
+using System.Net.Http;
+using System.Net.Http.Handlers;
+using System.Reflection;
 using System.Security.Cryptography;
 using System.Text;
 using System.Threading;
-using System.Reflection;
-using System.ComponentModel;
-using System.Collections.ObjectModel;
-using System.Linq;
-using Rampastring.Tools;
+using System.Threading.Tasks;
 using ClientUpdater.Compression;
+using Rampastring.Tools;
 
-namespace ClientUpdater
+public static class Updater
 {
-    public static class Updater
+    public const string SECOND_STAGE_UPDATER = "SecondStageUpdater.dll";
+    public static string VERSION_FILE = "version";
+    public const string ARCHIVE_FILE_EXTENSION = ".lzma";
+
+    /// <summary>
+    /// Currently set game path for the updater.
+    /// </summary>
+    public static string GamePath { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Currently set resource path for the updater.
+    /// </summary>
+    public static string ResourcePath { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Currently set local game ID for the updater.
+    /// </summary>
+    public static string LocalGame { get; private set; } = "None";
+
+    /// <summary>
+    /// Currently set calling executable file name for the updater.
+    /// </summary>
+    public static string CallingExecutableFileName { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Gets read-only collection of all custom components.
+    /// </summary>
+    public static ReadOnlyCollection<CustomComponent> CustomComponents => customComponents?.AsReadOnly();
+
+    /// <summary>
+    /// Gets read-only collection of all update mirrors.
+    /// </summary>
+    public static ReadOnlyCollection<UpdateMirror> UpdateMirrors => updateMirrors?.AsReadOnly();
+
+    /// <summary>
+    /// Update server URL for current update mirror if available.
+    /// </summary>
+    public static string CurrentUpdateServerURL => updateMirrors is { Count: > 0 } ?
+        updateMirrors[currentUpdateMirrorIndex].URL : null;
```

Comment:
```suggestion
    public static string CurrentUpdateServerURL
        => updateMirrors is { Count: > 0 }
            ? updateMirrors[currentUpdateMirrorIndex].URL
            : null;
```
IDK maybe a personal preference, but I'll leave it here as a suggestion